archaius-scala/src/main/scala/com/netflix/config/scala/ChainedBooleanProperty.scala

```diff
@@ -25,9 +25,12 @@ class ChainedBooleanProperty(
 extends ChainedProperty[Boolean]
 {
 
-  def this(prefix: Option[String], name: String, suffix: Option[String], default: Boolean, callback: Option[() => Unit] = None) = {
+  def this(prefix: Option[String], name: String, suffix: Option[String], default: Boolean, callback: Option[() => Unit]) = {
     this(ChainMakers.fanPropertyName(prefix, name, suffix), default, callback)
   }
+  def this(prefix: Option[String], name: String, suffix: Option[String], default: Boolean) = {
```

Comment:
**Question:** Why all of the additional signatures here? It's been a long while since I have written any Scala, did something change in more recent versions that makes this necessary?